### ISSUE TITLE
Identify appropriate vocabularies and ontologies for timeld content and schema

### ISSUE THREAD

**Issue by Angus-McAllister** [m-ld/timeld#1]
The m-ld timesheeting tool needs an RDF schema, whose terms should derive from upper and established ontologies wherever possible, and whose permitted values should align with widely-agreed controlled vocabularies.  Resolving the issue involves determining the appropriate ontologies and vocabularies for this purpose.

**Comment by Angus-McAllister** [comment]
Much research and deliberation has gone into this Issue since it was raised, out of which we have distilled some helpful tenets for making decisions about this.  These are:

1. **Cast the net wide** - there is a large number of ontologies and vocabularies out there, and in most cases, others who have encountered the same problem before, and come up with a solution.  This helps to avoid reinventing the wheel.
2. **Keep it simple** - it's tempting to latch onto the perfect term or concept from as many of the above ontologies as you come across, but this can also add unnecessary complexity.  If you have an idea of the semantics underpinning the data you're federating with - especially if some of those data are ontologically described - then align  to those.
3. **Couple loosely** - a big difference between data integration and data federation is that the latter is intentionally much more laissez-faire about the structure and meaning of the data, rather than seeking to coerce source data to conform to a schema firm specification.  This doesn't eliminate the need for data mapping and transformation, though (oh for a magic wand that can!) but is made a great deal easier (we hypothesize) through understanding the semantics of the data in question.  This late-binding approach broadens the range of systems to which one can federate, and gives more flexibility to those systems' maintainers to evolve their data models.

**Comment by Angus-McAllister** [comment]
So what have we actually delivered then?  Well, at the time of writing, there are now two iterations of the RDFS Class Model in this repo (see [architecture](https://github.com/m-ld/timeld/tree/main/architecture)), visualised as UML models.  UML isn't ideal for representing RDFS, since RDF Schema has aspects of both data and class modelling, which gets UML's knickers in a bit of a twist (especially if you're trying to apply it strictly).  However, it's the best that we've come across, and with a few compromises, does a reasonably good job of depicting it in a diagram.

The tool we used to create the diagrams is PlantUML (available as an IDE plugin), with the GraphViz helper.  It consumes text files (.puml) describing the model, using a pretty straightforward syntax, and enables the diagrams to be exported in a range of graphics file formats, including .svg, which is what we've used.
